### ISSUE TITLE
tsgo --build hangs on storybook tsconfig with ~100 story files (possible regression of #2678)

### ISSUE THREAD

**Issue by nicolas-besnard** [microsoft/typescript-go#3335]
### Environment
- `@typescript/native-preview`: `7.0.0-dev.20260403.1` (broken)
- Last working version: `7.0.0-dev.20260128.1`
- macOS Darwin 25.3.0

### Description

`tsgo --build --emitDeclarationOnly` hangs indefinitely when processing a `tsconfig.storybook.json` that includes ~108 `.stories.tsx` files using `@storybook/react-vite` types (`Meta<>`, `StoryObj<>`, decorators).

All upstream project references resolve as "up to date" — the hang occurs when tsgo begins typechecking the storybook project itself. `tsc` handles this fine.

This may be a regression of #2678 which was reported closed after "resolved with the latest main."

Also possibly related to #988 (same class of problem: React ecosystem complex generics at scale).

### Verbose output

```
[11:18:39 AM] Projects in this build:
    * ../utils/tsconfig.lib.json
    * ../models/tsconfig.lib.json
    * ...10 more referenced projects...
    * tsconfig.lib.json
    * tsconfig.spec.json
    * tsconfig.storybook.json
    * tsconfig.json

[11:18:39 AM] Project '../utils/tsconfig.lib.json' is up to date...
[11:18:39 AM] Project '../models/tsconfig.lib.json' is up to date...
...all dependencies up to date...
[11:18:39 AM] Project 'tsconfig.lib.json' is up to date...
[11:18:39 AM] Project 'tsconfig.spec.json' is up to date...
<-- hangs here, never prints status for tsconfig.storybook.json -->
```

### tsconfig.storybook.json

```json
{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "lib": ["DOM", "DOM.Iterable", "es2022"],
    "outDir": "out-tsc/storybook",
    "module": "esnext",
    "moduleResolution": "bundler",
    "jsx": "preserve"
  },
  "include": [
    "src/**/*.stories.ts",
    "src/**/*.stories.tsx",
    "src/**/*.stories.jsx",
    "src/**/*.stories.mdx",
    ".storybook/*.js",
    ".storybook/*.ts",
    ".storybook/preview.tsx"
  ],
  "references": [
    { "path": "./tsconfig.lib.json" }
  ],
  "files": [
    "../../node_modules/@nx/react/typings/styled-jsx.d.ts",
    "../../node_modules/@nx/react/typings/cssmodule.d.ts",
    "../../node_modules/@nx/react/typings/image.d.ts"
  ]
}
```

### Bisecting results

| Scenario | Result |
|---|---|
| `tsgo -p tsconfig.storybook.json` with all ~108 stories | **Hangs indefinitely** |
| `tsgo -p tsconfig.storybook.json` with 1 story | Completes instantly |
| `tsgo -p tsconfig.storybook.json` with ~10 stories | Completes instantly |
| `tsgo -p tsconfig.lib.json` (~200 component files, no stories) | Completes instantly |
| Same full config with `7.0.0-dev.20260128.1` | **Works fine** |

### Workaround

Pin `@typescript/native-preview` to `7.0.0-dev.20260128.1`.
